core/src/main/java/com/google/cloud/sql/core/DefaultConnectionInfoRepository.java

```diff
@@ -312,7 +312,7 @@ private InstanceMetadata fetchMetadata(CloudSqlInstanceName instanceName, AuthTy
             instanceName,
             ipAddrs,
             instanceCaCertificates,
-            "GOOGLE_MANAGED_CAS_CA".equals(instanceMetadata.getServerCaMode()),
+            isCasManagedCertificate(instanceMetadata),
```

Comment:
Would it simplify things to have this function take in `serverCaMode` instead of `instanceMetadata`?

Would make adding a standalone unit test for `isCasManagedCertificate` easy and reduce having to call `.getServerCaMode()` twice in the function.

```suggestion
            isCasManagedCertificate(instanceMetadata.getServerCaMode()),
```

Reply:
I've inverted the logic here. Going forward, both GOOGLE_MANAGED_CAS_CA, CUSTOMER_MANAGED_CAS_CA, and future new kinds of CA we will do standard TLS domain name validation using the server certificate SAN records. The certificate validation logic for the original `GOOGLE_MANAGED_INTERNAL_CA` is now the exception. The code should reflect it: 

We don't have a unit test for this method, but this change is covered by this test: `com.google.cloud.sql.core.ConnectorTest#create_successfulPublicCasConnection`

Reply:
I still think it would be simpler for the function to take in `serverCaMode` instead of `instanceMetadata`...

Reply:
Fixed.